firedrake/preconditioners/pmg.py

```diff
@@ -262,6 +275,11 @@ def coarsen_bcs(fbcs):
                           options_prefix=fctx.options_prefix,
                           transfer_manager=fctx.transfer_manager)
 
+        # TODO coarsen nullspace
```

Comment:
Should this TODO be addressed? Should we raise a `NotImplementedError`?

Reply:
I just implemented this. I will also include a test for this.